src/katgpucbf/xbgpu/engine.py

```diff
@@ -117,6 +121,426 @@ def reset(self, timestamp: int = 0) -> None:
+
+    def _populate_sensors(self) -> None:  # noqa: D102
+        sensors = self.engine.sensors
+        # Static sensors
+        sensors.add(
+            aiokatcp.Sensor(
+                str,
+                f"{self.output.name}.chan-range",
+                "The range of channels processed by this X-engine, inclusive",
+                default=f"({self.engine.channel_offset_value},"
+                f"{self.engine.channel_offset_value + self.engine.n_channels_per_stream - 1})",
+                initial_status=aiokatcp.Sensor.Status.NOMINAL,
+            )
+        )
+        # Dynamic sensors
+        sensors.add(
+            aiokatcp.Sensor(
+                bool,
+                f"{self.output.name}.rx.synchronised",
+                "For the latest accumulation, was data present from all F-Engines.",
+                default=False,
+                initial_status=aiokatcp.Sensor.Status.ERROR,
+                status_func=lambda value: aiokatcp.Sensor.Status.NOMINAL if value else aiokatcp.Sensor.Status.ERROR,
+            )
+        )
+        sensors.add(
+            aiokatcp.Sensor(
+                int,
+                f"{self.output.name}.xeng-clip-cnt",
+                "Number of visibilities that saturated",
+                default=0,
+                initial_status=aiokatcp.Sensor.Status.NOMINAL,
+            )
+        )
+
+    async def _flush_accumulation(self, tx_item: TxQueueItem, next_accum: int) -> TxQueueItem:
+        """Emit the current `tx_item` and prepare a new one."""
+        if tx_item.batches == 0:
+            # We never actually started this accumulation. We can just
+            # update the timestamp and continue using it.
+            tx_item.timestamp = next_accum * self.timestamp_increment_per_accumulation
+            return tx_item
+
+        # present_ants only takes into account batches that have
+        # been seen. If some batches went missing entirely, the
+        # whole accumulation is bad.
+        if tx_item.batches != self.output.heap_accumulation_threshold:  # type: ignore
+            tx_item.present_ants.fill(False)
+
+        # Update the sync sensor (converting np.bool_ to Python bool)
+        self.engine.sensors[f"{self.output.name}.rx.synchronised"].value = bool(tx_item.present_ants.all())
+
+        self.correlation.reduce()
+        tx_item.add_marker(self._proc_command_queue)
+        await self._tx_item_queue.put(tx_item)
+
+        # Prepare for the next accumulation (which might not be
+        # contiguous with the previous one).
+        tx_item = await self._tx_free_item_queue.get()
+        await tx_item.async_wait_for_events()
+        tx_item.timestamp = next_accum * self.timestamp_increment_per_accumulation
+        self.correlation.bind(out_visibilities=tx_item.buffer_device, out_saturated=tx_item.saturated)
+        self.correlation.zero_visibilities()
+        return tx_item
+
+    async def gpu_proc_loop(self) -> None:  # noqa: D102
```

Comment:
Ok so you haven't changed the `noqa` here but I suspect that's because you moved the docstring to the base class. Which is probably fine, but then this function will get nothing of its own in the module documentation.
That may be OK, I haven't considered the implications too deeply.

Reply:
Sphinx is configured to replicate the base class documentation in the derived class when it renders.